src/verifier.js

```diff
@@ -143,8 +163,22 @@ module.exports = function (options) {
 		throw new Error('Must provide both or none of --provider-states-url and --provider-states-setup-url.');
 	}
 
-	if (options.pactUrls) {
+	if (options.pactUrls && !checkTypes.emptyArray(options.pactUrls)) {
 		checkTypes.assert.array.of.string(options.pactUrls);
+		retrievePactsPromise = Promise.resolve(options.pactUrls);
```

Comment:
Still think that this promise should be created within the constructor since this function is more about validating input.